### DISCUSSION TITLE
Inconsistent Sentence Lengths

### DISCUSSION THREAD

**Discussion by egerber** [explosion/spaCy#6480] [Help: Coding & Implementations] [answered]
Hello,

I believe, I found some inconsistent behaviour of accumulated sentence lengths. I assume that sentence lengths should accumulate to the total document length. This, however, seems to be dependent on how many spaces I use between sentences.

## How to reproduce the behaviour
```python
import spacy

nlp=spacy.load('en_core_web_sm')
text="One.  Two.Three.     Four. Five"
doc=nlp(text)

accum_len=0
for sent in doc.sents:
    accum_len+=len(sent.text)
print(accum_len,len(text))
# 30, 31
print(len(doc.text),len(text))
# 31, 31

assert(len(doc.text)==len(text))
assert(accum_len==len(text))
```
But everything works fine for:
```python
text="One.  Two.Three.    Four.Five"
```

I hope you can reproduce this error.

Best,
Emanuel

## Info about spaCy

* **spaCy version:** 2.3.4
* **Platform:** Linux-5.4.0-53-generic-x86_64-with-debian-buster-sid
* **Python version:** 3.7.6

**Comment by adrianeboyd** [comment, marked as answer]
For both tokens and spans there may be a single trailing space that's not included in `token.text` and `span.text`. To get it with the trailing space, use `token.text_with_ws` or `span.text_with_ws` and then the lengths should work out as you expect.
